config/protobuf.gradle

```diff
@@ -0,0 +1,86 @@
+/**
+ * Adds simple protocol buffer compilation task that compiles protos from a "proto" folder of the
+ * module, into a "java_gen" folder. Code here is inspired by the Google protobuf gradle plugin,
+ * but is simpler and caused less issues with IDE integration and test execution.
+ */
+
+apply plugin: 'com.google.osdetector'
+
+/**
+ * The protocol buffer compiler and javalite plugin are pulled as dynamic dependencies, so that
+ * they do not need to be pre-installed by the developer.
+ */
+Configuration protoDepsConfig = project.configurations.create("protobufDeps") {
+    visible = false
+    transitive = false
+    extendsFrom = []
+}
+
+Dependency protocDep = project.dependencies.add(protoDepsConfig.name, [
+        group: "com.google.protobuf",
+        name: "protoc",
+        version: "3.2.0",
+        classifier: project.osdetector.classifier,
+        ext: 'exe'
+])
+
+Dependency javaliteDep = project.dependencies.add(protoDepsConfig.name, [
+        group: "com.google.protobuf",
+        name: "protoc-gen-javalite",
+        version: "3.0.0",
+        classifier: project.osdetector.classifier,
+        ext: 'exe'
+])
+
+/**
+ * Compiles the protocol buffers in the "proto" folder of the project, into the "java_gen" folder.
+ */
+task compileProtos {
+    doLast {
+        def protocPath = protoDepsConfig.fileCollection(protocDep).singleFile
+        def javalitePath = protoDepsConfig.fileCollection(javaliteDep).singleFile
+
+        def protoFiles = fileTree(dir: "${projectDir}/proto").matching { include "**/*.proto"}.sort()
+        logger.lifecycle("proto files: " + protoFiles)
+
+        def outDir = file("${projectDir}/java_gen")
+        if (!outDir.exists()) {
+            outDir.mkdir()
+        }
+
+        def cmd = [ protocPath ]
+        cmd += "--proto_path=${projectDir}/proto"
+        cmd += "--plugin=protoc-gen-javalite=" + javalitePath
+        cmd += "--javalite_out=${projectDir}/java_gen"
+        cmd.addAll(protoFiles)
+
+        logger.lifecycle("protoc command: " + cmd.toString())
+
+        def stdout = new StringBuffer()
+        def stderr = new StringBuffer()
+        Process compileProcess = cmd.execute()
```

Comment:
As noted offline, this appears to be failing on our CI https://www.bitrise.io/build/337d2c1b5567e59e